### PR TITLE
kvtenantccl: fix tenant upgrade test on new versions

### PR DESCRIPTION
**Pull request by AlexTalks**
While previously we had the `TestTenantUpgradeFailure` tests using a mix of
version references to `(min binary version, version-1, version)`,
this test fixes an issue (uncovered by the attempt to move to a new
version, 22.1.0) where we were referencing the version numbers
incorrectly, and standardizes the usage with a set of common variables
throughout the test. This allows us to avoid issues with `x.x.0`
releases, as in this case we are using `(min, version, version+1)`.

Release note: None

Release Justification: Testing fix.

### CODE REVIEW

**Comment by AlexTalks** [comment]
@celiala this should unblock #80502

File: pkg/ccl/kvccl/kvtenantccl/tenant_upgrade_test.go
Comment (irfansharif) [inline pkg/ccl/kvccl/kvtenantccl/tenant_upgrade_test.go:307]: This was the main change here unblocking the test, right? Looks like the commentary above also expected v1, so wonder when we regressed.

**Review by irfansharif — APPROVED** [review]

**Review by AlexTalks — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80606)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @fqazi and @irfansharif)

___
*[`pkg/ccl/kvccl/kvtenantccl/tenant_upgrade_test.go` line 307 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/80606#-N0cXPAzA3DosjujTQDP-r1-307:-N0gPbJO9awKcexN3raG:bm0ronj) ([raw file](https://github.com/cockroachdb/cockroach/blob/d53d13dcf258eba2cde19e09a4ce8164d871548c/pkg/ccl/kvccl/kvtenantccl/tenant_upgrade_test.go#L307)):*
<details><summary><i>Previously, irfansharif (irfan sharif) wrote…</i></summary><blockquote>

This was the main change here unblocking the test, right? Looks like the commentary above also expected v1, so wonder when we regressed.
</blockquote></details>

To be honest I think this never quite worked properly - I don't think it was a regression, but not sure.  The usages of `clusterversion.TestingBinary[MinSupported]Version` and `v0, v1, v2` were quite mixed throughout the test.

It looks like previously it set the cluster version to `TestingBinaryMinSupportedVersion (v0)`, with the binary version at `TestingBinaryVersion (v2)`, attempts to upgrade directly from `v0 -> v2` but doesn't crash before our `SELECT` since we don't read off `tenantStopperChannel` until after (and I guess the `v1 -> v2` migration doesn't block the tenant from accepting queries?), and then we force the crash, check that we're still at `v1`, and move on.

I guess all that is to say - we also really need to move the `go func() { <-tenantStopperChannel ...etc}` to start prior to the first `SET CLUSTER SETTING`.  Can amend this commit to do that here.

**Review by celiala — APPROVED** [review]
Thank you, Alex!

I'll let others review the PR for correctness. But approving here to denote that this PR unblocks https://github.com/cockroachdb/cockroach/pull/80502 (rebased on top of this PR to confirm all tests pass)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80606#-:-N0gaFlL7misyOKjcUgl:banrlui)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @fqazi and @irfansharif)

**Review by AlexTalks — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80606)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @fqazi and @irfansharif)

___
*[`pkg/ccl/kvccl/kvtenantccl/tenant_upgrade_test.go` line 307 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/80606#-N0cXPAzA3DosjujTQDP-r1-307:-N0h0l5C1B1WsLt_g_Y1:b-qv62i) ([raw file](https://github.com/cockroachdb/cockroach/blob/d53d13dcf258eba2cde19e09a4ce8164d871548c/pkg/ccl/kvccl/kvtenantccl/tenant_upgrade_test.go#L307)):*
<details><summary><i>Previously, AlexTalks (Alex Sarkesian) wrote…</i></summary><blockquote>

To be honest I think this never quite worked properly - I don't think it was a regression, but not sure.  The usages of `clusterversion.TestingBinary[MinSupported]Version` and `v0, v1, v2` were quite mixed throughout the test.

It looks like previously it set the cluster version to `TestingBinaryMinSupportedVersion (v0)`, with the binary version at `TestingBinaryVersion (v2)`, attempts to upgrade directly from `v0 -> v2` but doesn't crash before our `SELECT` since we don't read off `tenantStopperChannel` until after (and I guess the `v1 -> v2` migration doesn't block the tenant from accepting queries?), and then we force the crash, check that we're still at `v1`, and move on.

I guess all that is to say - we also really need to move the `go func() { <-tenantStopperChannel ...etc}` to start prior to the first `SET CLUSTER SETTING`.  Can amend this commit to do that here.
</blockquote></details>

I may have been wrong about this never working properly - I think I was a bit confused about how the upgrade was actually working.  The _host_ cluster should be upgraded to the latest binary version available, as upgrading the host cluster to latest+1 (I.e. 22.1-2) when that version doesn't actually exist will never complete successfully.  However, upgrading the host cluster doesn't perform the tenant upgrade migrations, which is where we actually force the stop.  I.e. what we want to do is:

- upgrade host cluster to latest real version
- try to upgrade tenant to `v2` (which is either latest, or latest+1 in the case of a x.x.0)
- see that this fails.

We should maybe have better logic so that we always have `v1` as version-1, but I'll leave that for a future change.

**Pull request merged**
